[Triton][Inductor] Infer Boolean Types

csteegz opened the pull request:
Summary: PT2 compiler has issues with boolean types in wrapped functions. There is some code to try to infer if an unknown type is an i32 or i64, but that causes a failure when it tries to compare with a boolean. Add explicit tests to determine if data is `i1`.

Test Plan:
Added test to test_triton_kernels.py

I'm having trouble figuring out how to run locally but expect the unit test to work with existing infrastracture. 

I have run locally and verified a wrapped triton kernel can be compiled.

Differential Revision: D69805822




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

oulgen reviewed (commented):
Code looks good, but please fix the lint errors

csteegz commented on test/inductor/test_triton_kernels.py:
```diff
@@ -2117,6 +2117,60 @@ def fn(x):
+        # The clamp_one input needs to be marked as "i1".
+        @triton.jit
+        def triton_add_noise_(
+            x_ptr, y_ptr, seed, numel, clamp_one, BLOCK_SIZE: tl.constexpr
```
no, to be honest I haven't. Let me try with `tl.constexpr` and if it works I'll switch to that.

csteegz committed:
[Triton][Inductor] Infer Boolean Types (#147416)

Summary:

PT2 compiler has issues with boolean types in wrapped functions. There is some code to try to infer if an unknown type is an i32 or i64, but that causes a failure when it tries to compare with a boolean. Add explicit tests to determine if data is `i1`.

Test Plan:
Added test to test_triton_kernels.py

I'm having trouble figuring out how to run locally but expect the unit test to work with existing infrastracture.

I have run locally and verified a wrapped triton kernel can be compiled.

Reviewed By: sijiac

Differential Revision: D69805822
